packages/next/src/build/webpack/plugins/next-types-plugin.ts

```diff
@@ -326,17 +348,22 @@ export class NextTypesPlugin {
     this.pageExtensions = options.pageExtensions
     this.pagesDir = path.join(this.appDir, '..', 'pages')
     this.typedRoutes = options.typedRoutes
-    this.originalRewrites = options.originalRewrites
-    this.originalRedirects = options.originalRedirects
+    if (this.typedRoutes && !redirectsRewritesRoutesProcessed) {
+      redirectsRewritesRoutesProcessed = true
+      processRedirectsRewritesRoutes(
+        options.originalRewrites,
+        options.originalRedirects
+      )
+    }
   }
 
   collectPage(filePath: string) {
     if (!this.typedRoutes) return
 
     const isApp = filePath.startsWith(this.appDir + path.sep)
 
-    // Filter out non-page files in app dir
-    if (isApp && !/[/\\]page\.[^.]+$/.test(filePath)) {
+    // Filter out non-page and non-route files in app dir
+    if (isApp && !/[/\\](?:page|route)\.[^.]+$/.test(filePath)) {
```

Comment:
Let's avoid route for now (because it's only a valid route if it exports GET), we can discuss that on the other issue thread.

Reply:
I've reverted the change :)